snuba/datasets/cdc/groupedmessage_processor.py

```diff
@@ -1,54 +1,88 @@
+from datetime import datetime
+from typing import Any, Mapping, Optional
+
+from dataclasses import dataclass
 from dateutil.parser import parse as dateutil_parse
-from snuba.datasets.cdc.cdcprocessors import CdcProcessor
+from snuba.datasets.cdc.cdcprocessors import CdcProcessor, CDCMessageRow
 
 
-class GroupedMessageProcessor(CdcProcessor):
+@dataclass(frozen=True)
+class GroupMessageRecord:
+    status: int
+    last_seen: datetime
+    first_seen: datetime
+    active_at: Optional[datetime] = None
+    first_release_id: Optional[int] = None
 
-    def __init__(self, postgres_table):
-        super(GroupedMessageProcessor, self).__init__(
-            pg_table=postgres_table,
-        )
 
-    def __build_record(self, offset, columnnames, columnvalues):
+@dataclass(frozen=True)
+class GroupedMessageRow(CDCMessageRow):
```

Comment:
This wrapper around the record data could probably be made into a generic later but probably best not to worry about that until there is a use case for it

Reply:
The issue is the identity key. Just the fact that soon we will add project to the GroupedMessage makes it not generic anymore. Unless we make the parent class a "Generic" with identity as parameter. Which is something we can consider.